### ISSUE TITLE
Package request: moonrepo

### ISSUE THREAD

**Issue by darkone23** [NixOS/nixpkgs#232712]
**Project description**

Build system for monorepo projects

**Metadata**

* homepage URL: https://moonrepo.dev/
* source URL: https://github.com/moonrepo/moon
* license: mit
* platforms: linux, etc

Comments:

Cargo project, github releases ships with dynamic binaries but I am not sure how to use those in nixOS...

**Comment by dit7ya** [comment]
https://github.com/NixOS/nixpkgs/pull/229867

**Comment by darkone23** [comment]
Thank you for packaging this. 

I was working on a weird project flake locally and I ran into those test issues as well. 

Now I am wondering... perhaps somehow moon is incompatible with nixos because of those test failures?

I opened this issue on the project:
https://github.com/moonrepo/moon/issues/858

And I discovered those tests were failing because of unpatched node binaries trying to run stuff.

I wanted to work on a project that uses moon to manage its dependencies.

Do those node failures mean that the way moon is trying to help manage frontend dependencies is broken on nixos?
